plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -809,14 +809,18 @@ func (checker *PodAffinityChecker) InterPodAffinityMatches(pod *api.Pod, meta in
 
 	// Check if the current node match the inter-pod affinity scheduling constraints.
 	// Hard inter-pod affinity is not symmetric, check only when affinity.PodAffinity exists.
-	if affinity.PodAffinity != nil {
+	if affinity != nil && affinity.PodAffinity != nil {
```

Comment:
I think the code from here to the end of the function would be more compact as

I think this would be more compact like
```
if affinity != nil {
  if affinity.PodAffinity != nil && !check.NodeMatchesHardPodAffinity(...) {
         return false, ErrPodAffinityNotMatch
   }
   if affinity.PodAntiAffinity != nil && !check.NodeMatchesHardPodAffinityAntiAffinity(...) {
        return false, ErrPodAffinityNotMatch
   }
}
return true, nil
}